### PR TITLE
docs: Publish website

### LINKED ISSUE #26: Cleanup the Readme and setup docs website with Docusaurus
There's a lot of information in the Readme right now. We should move the API and examples into separate files and then maybe generate a website out of it with Docusaurus.

### PR DESCRIPTION
**Pull request by dcalhoun**
### Summary

* Remove unused Docusaurus code.
* Add CI task for publishing website.

Fixes #26.

---

The CI task is the basic one provided by the [Docusaurus guides](https://docusaurus.io/docs/en/publishing#using-circle-ci-20). It has been awhile since I've interfaced with GitHub pages, it's also difficult to test without access to the repository. You may need to configure it further, but hopefully this provides a good starting place.

### Test plan

1. Merge this work.
1. Configure CircleCI with GitHub token.
1. Commit to `master` to trigger build and deployment.

### CODE REVIEW

File: website/.circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Circle CI 2.0 Config File
```
Comment (thymikee) [inline website/.circleci/config.yml]: this config file is not necessary, can you remove it?

Comment (dcalhoun) [reply website/.circleci/config.yml]: @thymikee removed.

**Comment by thymikee** [comment]
Ok, let's give it a try!

**Comment by thymikee** [comment]
Works! https://callstack.github.io/react-native-testing-library/
Thank you so much for making that happen ❤️. If you have ideas for improvements (the website is pretty raw at this point) feel free to share them 👍

**Comment by dcalhoun** [comment]
🎉 Awesome! I'll let you know if I encounter issues or have ideas for improvements. If your team produces a custom design, let me know and I may be able to find time to help execute it.

**Pull request merged**